compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc

```diff
@@ -3430,23 +3430,34 @@ INTERCEPTOR(uptr, ptrace, int request, int pid, void *addr, void *data) {
   COMMON_INTERCEPTOR_ENTER(ctx, ptrace, request, pid, addr, data);
   __sanitizer_iovec local_iovec;
 
-  if (data) {
+  void *data_arg = data;
+#  if defined(__sparc__)
+  // The meanings of addr and data are reversed for a few requests on
+  // Linux/sparc64.
+  if (request == ptrace_getregs || request == ptrace_getfpregs ||
```

Comment:
That's really weird.
Can you add a comment with proof-link here for future readers?

Reply:
Done: it's on `ptrace(2)`.